packages/lisk-transaction-pool/src/job.ts

```diff
@@ -0,0 +1,30 @@
+export class Job<T> {
+	// tslint:disable-next-line variable-name
+	private _id: NodeJS.Timer | undefined;
+	// tslint:disable-next-line variable-name
+	private readonly _interval: number;
+	// tslint:disable-next-line variable-name
+	private readonly _job: () => Promise<T>;
+
+	public constructor(
+		context: object,
+		job: () => Promise<T>,
+		interval: number,
+	) {
+		this._interval = interval;
+		this._job = job.bind(context);
+	}
+
+	public start(): void {
+		if (typeof this._id === 'undefined') {
+			this._id = setInterval(this._job, this._interval);
```

Comment:
What if a job didn't finish yet? Does the function suppose to start jobs regardless of the results?
 
Let's say if a function execution lasts for 200ms and the interval is set to 100ms, the call stack will be exceeded sooner or later. 

There is a file of identical purpose in Lisk Core - `jobs_queue.js` which uses setTimeout. 

It would be better to keep one file and share it across the projects, rather than creating two of similar behaviours.

Reply:
I agree that maybe it's important to use setTimeout instead of setInterval while I think it's important to have metrics for the execution time of jobs. It's important because the transaction pool should fire events at certain times before forging blocks etc.

In lisk-elements, we aren't using jobs anywhere else as of yet. So, there is no code duplication inside the project. I don't know how we can share functionality between Core and SDK. We can develop a strategy, later on, to share common utility functions/classes between projects. But until then, I think it's okay to have it here. What do you think?

Reply:
@shuse2, I agree that we should devise a strategy for calculating metrics. I've created a new issue: #937. In this issue, we can discuss and finalize the approach we want to follow to calculate the performance of the new transaction pool as a whole.